### PR TITLE
Use new api command AddWorkspaceRoom

### PR DESCRIPTION
**Pull request by neil-marcellini**
cc @jasperhuangg since you reviewed the Web-PR and @tgolen I chatted with you briefly about adding an extra OfflineIndicator.

### Details

`AddWorkspaceRoom` creates an optimistic room and navigates to it. If the user is offline, the same thing happens but the whole page is grayed out until they come back online. You can still send chats in the room while offline. 

Front end validation should prevent any errors creating rooms, but it is possible that a room with a duplicate name could be created. In that case, we gray out the page, remove the composer, and show an error message. When the error is dismissed the user will be navigated to the concierge chat and the room and its actions (chats) will be deleted.

I also fixed two additional issues with this PR. The first issue was fixed by removing the `actorEmail` for created actions on rooms. The second issue was fixed by adding a separate `OfflineIndicator` to the ReportScreen for large screens when the composer is hidden. See the issues for more details. 

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/215186
$ https://github.com/Expensify/App/issues/10948
$ https://github.com/Expensify/App/issues/10895

### Tests

**Online**
1. Log in to any account A
2. Create a workspace by clicking the green plus, New workspace (if needed)
3. Using another platform or incognito window, log in to a different account B
4. As user A, invite user B to your workspace by clicking on the use avatar in the header, workspace name, Manage members, Invite, enter user B's email, select them, and click Invite.
5. Click the green plus then click New room
6. Fill out the new room form as a restricted workspace and click save
7. Verify that you are navigated to the newly created room and you see the "Collaboration starts here" message.
8. Click the room header bar and make sure you can view the details.
9. As user B, click the search icon in the header
10. Enter the room name, make sure it shows up, and click on it
11. Verify that the room opens
12. Send a message
13. As user A, verify that you receive the message

**Offline**

1. Log in to any account
2. Create a workspace by clicking the green plus, New workspace (if needed)
3. Go offline
4. Click the green plus then click New room
5. Fill out the new room form as a restricted workspace and click save
6. Verify that you are navigated to the newly created room and you see the "Collaboration starts here" message and everything is grayed out
7. Go back online
8. Verify that the room page is no longer grayed out.

**Errors**

1. Comment out the front end report name validation here https://github.com/Expensify/App/blob/70bc06e9d5bcffeefc374fc4f23c8e7f60cdf2b8/src/pages/workspace/WorkspaceNewRoomPage.js#L83-L85
2. Log in to any account
3. Create a workspace by clicking the green plus, New workspace (if needed)
4. Click the green plus then click New room
5. Fill out the new room form with a name like `test-duplicate-name` as a restricted workspace and click save
6. Verify that you are navigated to the newly created room and you see the "Collaboration starts here" message.
7. Repeat steps 4-6 to create a duplicate room
8. Verify that the RBR and error shows
9. Hit the X to dismiss the error
10. Verify that you are navigated to another chat and the room no longer exists

**For #10948**
1. Sign into an account
2. Go offline
3. Create a workspace. If you don't have any workspaces, click the green plus, New workspace. Otherwise, open settings by clicking the avatar in the header, click on a workspace, click the 3 dots in the header, New workspace.
4. Close the settings
5. Click the search bar in the header
6. Type announce in the search bar
7. Click on the #announce room for the workspace you just created
8. Send a message
9. Verify that a message appears with your avatar and name
Note that creating a workspace while offline is currently broken. That is a separate issue reported here https://github.com/Expensify/App/issues/10897.

**For #10895**
1. Sign into any account on a large screen (Web or desktop).
2. Click on an archived room in the side bar and skip to step 6. If you don't have an archived room do the steps below to create one.
3. Create a workspace by clicking the green plus, New workspace (if needed)
4. Open the settings by clicking the avatar in the header, workspace name, 3 dots in the header, delete workspace
5. Click the search bar in the header, enter announce and click on the archived announce room
6. Go offline
7. Verify that the offline indicator should appear

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

All tests above except for the **Errors** test.

- [ ] Verify that no errors appear in the JS console

#### Web

Online success
image

Creating a room offline, commenting, and editing the comment.
image
image
image

Error
image

#### Mobile Web

**iOS / Safari**
Online success

Error

**Android / Chrome**
Online success

Error

#### Desktop

Online success
image
Error
image

#### iOS

Online success

Error

#### Android

Success

Error

### CODE REVIEW

File: src/libs/actions/Report.js
Comment (neil-marcellini) [inline src/libs/actions/Report.js:1476]: I want to point this comment out because it's interesting and a little funky.

**Comment by PauloGasparSv** [comment]
Hey @neil-marcellini, I mentioned the issue https://github.com/Expensify/App/issues/11074 here since I don't know if the bug will remain after this refactor.

**Comment by neil-marcellini** [comment]
While testing with the simulator on mobile web I encountered this navigation glitch. There is an open issue for that here https://github.com/Expensify/App/issues/8292.

Here is a video of the problem.
https://user-images.githubusercontent.com/26260477/191092370-aa82d608-77f0-4e06-a08d-b702cdeed5e8.mp4

File: src/libs/actions/Report.js
Comment (tgolen) [inline src/libs/actions/Report.js:1460]:
```suggestion
function addPolicyReport(policy, reportName, visibility) {
```
Can you please update this everywhere? Full context is in this thread: https://expensify.slack.com/archives/C02HWMSMZEC/p1661522786746159

**Review by tgolen — CHANGES_REQUESTED** [review]

File: src/libs/actions/Report.js
Comment (neil-marcellini) [reply src/libs/actions/Report.js:1460]: Ah right, I should have remembered. I'll update now.

**Comment by neil-marcellini** [comment]
Ready for more reviews!

**Review by tgolen — APPROVED** [review]

File: src/libs/OptionsListUtils.js
Comment (jasperhuangg) [inline src/libs/OptionsListUtils.js:500]: NAB: this is definitely a complex and difficult to understand condition, I wonder if there are any ways to simplify it

File: src/pages/home/report/ReportActionItemCreated.js
Comment (jasperhuangg) [inline src/pages/home/report/ReportActionItemCreated.js:51]: any chance report could be `null` here? I'm personally not sure but to be safe let's use `lodashGet`?

**Review by jasperhuangg — COMMENTED** [review]
Overall lookin good, few super tiny concerns

File: src/libs/OptionsListUtils.js
Comment (neil-marcellini) [reply src/libs/OptionsListUtils.js:500]: Yeah that's pretty ugly. It would help to break it out into several descriptive boolean variables. I think @tgolen might be cleaning that up with his LHN cleanup issues? I'm going to leave it as is for now.

File: src/pages/home/report/ReportActionItemCreated.js
Comment (neil-marcellini) [reply src/pages/home/report/ReportActionItemCreated.js:51]: `report` will always be non-null because of `withOnyx`. It will not render the component until the report is read https://github.com/Expensify/react-native-onyx/blob/bac8d308cb2eaf971174837e30a1f27f6b54c71e/lib/withOnyx.js#L159-L161

File: src/libs/OptionsListUtils.js
Comment (neil-marcellini) [reply src/libs/OptionsListUtils.js:500]: Also, this logic has actually been moved here in main
https://github.com/Expensify/App/blob/7fbe704ac8291944a7a8533bbcfa1df0e207b182/src/libs/SidebarUtils.js#L125-L134

**Comment by neil-marcellini** [comment]
I think I have addressed all your comments @jasperhuangg. I'll wait for your approval and then I think we're good to merge!

File: src/pages/home/report/ReportActionItemCreated.js
Comment (tgolen) [reply src/pages/home/report/ReportActionItemCreated.js:51]: There might still be something to be careful of here though. 

> It will not render the component until the report is read

While this is true, it does not guarantee that a report is returned. It could read the report from Onyx, find nothing there, then pass `null` as a prop. In fact, you see that `report` is not a required prop.

File: src/libs/OptionsListUtils.js
Comment (tgolen) [reply src/libs/OptionsListUtils.js:500]: Yeah, let me take a try at cleaning up this logic 👍 I created a GH for it here: https://github.com/Expensify/Expensify/issues/229633

File: src/pages/home/report/ReportActionItemCreated.js
Comment (neil-marcellini) [reply src/pages/home/report/ReportActionItemCreated.js:51]: Oh yeah, good point. `report` is a required prop of ReportActionList, which renders the action items and in turn this component, so I think it's safe. ReportActionList also accesses `props.report.reportID` without any checks.

**Comment by jasperhuangg** [comment]
Loooookin good :)

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

**Review by jasperhuangg — APPROVED** [review]

**Comment by jasperhuangg** [comment]
https://github.com/Expensify/Web-Expensify/pull/34745 in on prod, so taking this off HOLD and merging!

**Comment by neil-marcellini** [comment]
Last set of checks passed.

File: src/libs/actions/Report.js
Comment (francoisl) [inline src/libs/actions/Report.js:1524]: Coming from https://github.com/Expensify/App/issues/20502 -  it would have been best to add a `failureData` here, otherwise if the backend fails to send a proper onyxUpdate in case of an error, the NewDot client won't know that the room wasn't created.

**Pull request merged**
